app/src/editor/shared/utils/bounds.js

```diff
@@ -120,3 +120,20 @@ export const findNonOverlappingPosition = (
 
     return findNonOverlappingPosition(nextBB, boundingBoxes, stackOffset, false)
 }
+
+export function findNonOverlappingPositionForModule(canvas: any, moduleData: any): any {
+    const moduleBounds = {
+        x: parseInt(moduleData.layout.position.left, 10) || 0,
+        y: parseInt(moduleData.layout.position.top, 10) || 0,
+        width: parseInt(moduleData.layout.width, 10) || 150,
+        height: parseInt(moduleData.layout.height, 10) || 100,
+    }
+    const boundingBoxes = canvas.modules.map((m) => getModuleBounds(m))
```

Comment:
`canvas.modules.map(getModuleBounds)` dot dot dot. 💃

Reply:
I generally avoid this because weird things can happen if `getModuleBounds` ever gets a second or third argument.

Reply:
Makes sense. Somehow I feel that Flow should keep an eye on such things… but we all know how it is with flow and its fairly flexible type system.

Reply:
also wouldn't help if the second argument happened to be an unrelated `number`